# ==============================================================================
# FIXING COMMIT DIFF
# ==============================================================================
# File: js/src/jit-test/tests/gc/bug-1757573.js
# Commit: d30323c51a42
# Full Hash: d30323c51a42996c0cbab0597f15b7abf57b720a
# Author: Jon Coppeard <jcoppeard@mozilla.com>
# Date: 2022-03-04 15:59:45
# Description:
#   Bug 1757573 - Use the correct zone when looking up nuked WeakRef CCWs r=sfink
#   
#   WeakRefs are exposed to the user so anyone can create CCWs to WeakRefs. Don't
#   assume that a nuked WeakRef CCW is one that we created but get it's target to
#   find which zone to look it up in.
# ==============================================================================

diff -r 4bcb69527b98 -r d30323c51a42 js/src/jit-test/tests/gc/bug-1757573.js
--- /dev/null	Thu Jan 01 00:00:00 1970 +0000
+++ b/js/src/jit-test/tests/gc/bug-1757573.js	Fri Mar 04 10:08:31 2022 +0000
@@ -0,0 +1,7 @@
+new FinalizationRegistry(a => {});
+b = newGlobal({newCompartment: true});
+b.eval(`
+  c = {};
+  d = new WeakRef(c);
+`);
+nukeCCW(b.d);
